### ISSUE TITLE
The add-on does not work after the NVDA update endpoint API migration.

### ISSUE THREAD

**Issue by cary-rowen** [nvda-es/updateChannel#16]
For some backgrounds, see the following:
https://github.com/nvaccess/nvda/issues/17686

## Steps to reproduce
1. Install NVDA2024.4.2 and UpdateChennel add-ons.
2. Change the update channel to RC or Alpha.
3. Go to the NVDA's Help menu and check for updates.

## Actual behavior
NVDA has not checked for any updates.
### Expected behavior# 预期行为
NVDA should discover new versions of the corresponding channel.

**Comment by jmdaweb** [comment]
Hi,
By default, this add-on doesn't modify endpoint URLs, but only a variable on versionInfo module which determines which channel will be used. I can patch NVDA and use new endpoints if required, but before that I need to know if old ones are definitely deprecated or will be kept to be used in older NVDA versions or when updating from 2024.4 to 2025.1.
Regards.

**Comment by cary-rowen** [comment]
Hi @jmdaweb 
I don't know what caused the problem yet. But the add-on works as expected before NV Access migrates the API endpoint.
Although NV Access enabled new updateCheck API endpoints, 301 redirect are set for older API at the same time.

In 2025.1, the new API endpoint will be used:
https://github.com/nvaccess/nvda/pull/17680

**Comment by cary-rowen** [comment]
Any ideas?

**Comment by GeorgeWu3** [comment]
I cannot update to the latest RC or the latest Alpha using the update channel selector add on.

**Comment by GeorgeWu3** [comment]
Same here, I’m not sure why is it not working.

**Comment by jmdaweb** [comment]
OK, now I can reproduce the issue. Seems the new endpoints perform additional checks which go beyond a simple variable, which this add-on was changing to switch between channels. @javidominguez can you please help me to investigate wether this issue has a solution?

**Comment by jmdaweb** [comment]
After a short investigation, the results are as follows:

* Alpha versions can upgrade/downgrade to latest alpha, beta or stable. All channels work.
* Stable versions can upgrade to latest beta or stable version, depending on the selected channel.
* Beta versions can upgrade to latest beta, or latest stable release when it's newer than the beta. Server prevents downgrading from a beta to the previous stable, for example from 2025.1rc1 to 2024.4.2.

Thush, the add-on is working perfectly and there are no problems on this side. You should discuss this problem with NV Access on the general mailing list, or by opening an issue on the NVDA repository.
Closing as completed.

**Comment by cary-rowen** [comment]
cc @jmdaweb 
I have to say that this add-on is still not work for me:
The current 2025.1 stable cannot check 2025.2Alpha, although I have set the update channel to Alpha.

**Comment by jmdaweb** [comment]
Extending my comment above: an alpha version can't be installed from a beta or stable one. This is related to the server. After a private conversation with NV Access via e-mail, I have decided this won't change. Sorry. The add-on is not as practical as before. Documentation will be updated soon.

**Comment by GeorgeWu3** [comment]
I guess we have to accept the new changes for the update channel selector
add-on for NVDA. It is not good to downgrade your copy of NVDA.
